### PR TITLE
[Tests] [pallet-timestamp] Add tests for OnTimestampSet.

### PR DESCRIPTION
**Pull request by TarekkMA**
Add tests for `OnTimestampSet` for the pallet `timestamp`.

There should be no change in the API, just refactoring code to be in separate files and adding a new test.

polkadot: 12xH9F4aSGEq27AZxLqG15fheDTtSxh1rJkVSXQ2g59djs7u

### CODE REVIEW

**Comment by ggwpez** [comment]
It needs a format to make the CI happy.

**Comment by TarekkMA** [comment]
Okay will check that

File: frame/timestamp/src/mock.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+use super::*;
```
Comment (bkchr) [inline frame/timestamp/src/mock.rs:1]: Missing license header.

File: frame/timestamp/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+use frame_support::assert_ok;
```
Comment (bkchr) [inline frame/timestamp/src/tests.rs:1]: License header missing as well.

**Review by bkchr — COMMENTED** [review]
In general I don't get why we need mock.rs? What is the benefit? Currently I don't see one and we could just use the test.rs

**Comment by TarekkMA** [comment]
@bkchr Thank you for the review, I've introduced `mock.rs` to contain the code related to the testing setup. (following the pattern of other pallets in the substrate codebase).

so `tests.rs` will only house tests.

**Comment by TarekkMA** [comment]
@ggwpez the formatting issues are not in the scope of this PR. I will see if I can open a new PR with this fix

**Comment by ggwpez** [comment]
I'm not sure if this is a useful change TBH, but as long as it is correct is there a reason not to merge it?

**Comment by TarekkMA** [comment]
Thank you.

Noted, will not use force push in my future contributions.

**Comment by ggwpez** [comment]
bot merge

**Comment by TarekkMA** [comment]
> /tip small
@shawntabrizi Thank you for the tip ❤️. seems like the bot didn't catch this comment.

**Comment by shawntabrizi** [comment]
@TarekkMA let me figure out whats up

**Comment by TarekkMA** [comment]
Thank you much appreciated

**Pull request merged**